codexctl/device.py

```
@@ -274,11 +286,140 @@ def connect_to_device(
+
+    def _get_paper_pro_partition_info(self, current_version: str) -> tuple[int, int, int]:
+        """Gets partition information for Paper Pro devices
+
+        Args:
+            current_version: Current OS version string for version-aware detection
+
+        Returns:
+            tuple: (current_partition, inactive_partition, next_boot_partition)
+        """
+        if not self.client:
+            raise RuntimeError("SSH client required for partition detection")
```

Ruff (0.14.5)

Avoid specifying long messages outside the exception class

(TRY003)